src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -245,6 +245,56 @@ void sendBinaryMessageWithKey(VertxTestContext context) throws InterruptedExcept
         });
     }
 
+    @Disabled("Will be check in the next PR, this is just external tests for Bridge")
+    @Test
+    void sendTextMessageWithKey(VertxTestContext context) throws InterruptedException, ExecutionException {
+        KafkaFuture<Void> future = adminClientFacade.createTopic(topic, 2, 1);
+
+        String value = "message-value";
+        String key = "my-key-text";
+
+        JsonArray records = new JsonArray();
+        JsonObject json = new JsonObject();
+        json.put("value", value);
+        json.put("key", key);
+        records.add(json);
+
+        JsonObject root = new JsonObject();
+        root.put("records", records);
+
+        future.get();
+
+        producerService()
+            .sendRecordsRequest(topic, root, BridgeContentType.KAFKA_JSON_TEXT)
+            .sendJsonObject(root, verifyOK(context));
+
+        Properties consumerProperties = Consumer.fillDefaultProperties();
+        consumerProperties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaUri);
+
+        KafkaConsumer<byte[], byte[]> consumer = KafkaConsumer.create(vertx, consumerProperties,
+                new ByteArrayDeserializer(), new ByteArrayDeserializer());
+        consumer.handler(record -> {
+            context.verify(() -> {
+                assertThat(new String(record.value()).replace("\\", ""), is(value));
```

Comment:
```suggestion
                assertThat(new String(record.value()), is(value));
```

Reply:
As I said before, it was a trying. 

There is \ though. 

Did you write any test in your fork?

Reply:
> There is \ though.

Nope. There is not. I am not receiving any `\' when running the test and it works fine. There is no need to use the `replace` because `\` is not in the received message.

> Did you write any test in your fork?

No I am not. The PR is yours :-)

Reply:
Why there is no test for the binary format?

Reply:
I don't know, maybe was planned and not done, but it's not subject for this PR :-)

Reply:
> I don't know, maybe was planned and not done, but it's not subject for this PR :-)

Seems like it was not completed for some reason. 

The thing that is happening in all tests here is the same. As explained [here](https://github.com/strimzi/strimzi-kafka-bridge/pull/836#issuecomment-1869509917)

Reply:
Tbh I am not following you. All tests are green in the main branch. Not sure you broke something on yours.

Reply:
> In the main branch, the tests for the binary embedded format are ignored, if you manually run them, it will probably fail.

Ok, but it's totally unrelated to your PR and what you are trying to achieve. I don't follow why you are mentioning this more times.

Reply:
> > In the main branch, the tests for the binary embedded format are ignored, if you manually run them, it will probably fail.
> 
> Ok, but it's totally unrelated to your PR and what you are trying to achieve. I don't follow why you are mentioning this more times.

Yes, I know! I was just wondering why. 
Anyway, I had to open a fresh PR with all changes and tests now passing.
Please take a look :)